### PR TITLE
Start wilms analysis

### PR DESCRIPTION
**Pull request by maud-p**
### Purpose/implementation Section
In this module 1, I create 2 metadata tables to compile from the literature information on marker genes and known genetic alterations, that will be used later to validate annotations of the Wilms tumor dataset. 

#### Please link to the GitHub issue that this pull request addresses.
https://github.com/AlexsLemonade/OpenScPCA-analysis/issues/671
https://github.com/AlexsLemonade/OpenScPCA-analysis/discussions/635#discussioncomment-10140478

#### What is the goal of this pull request?

Wilms tumor (WT) is the most common pediatric kidney cancer characterized by an exacerbated intra- and inter- tumor heterogeneity. The genetic landscape of WT is very diverse in each of the histological contingents. The COG classifies WT patients into two groups: the favorable histology and diffuse anaplasia. Each of these groups is composed of the blastemal, epithelial, and stromal populations of cancer cells in different proportions, as well as cells from the normal kidney, mostly kidney epithelial cells, endothelial cells, immune cells and normal stromal cells (fibroblast).

In this module, we reviewed the literature to compile a table of marker genes for each of the expected cell types in the dataset. Additionally, we provide a table of know genetic alterations in Wilms tumor that can be useful to validate CNV profiles obtained after running inferCNV function.

#### Briefly describe the general approach you took to achieve this goal.

The table CellType_metadata.csv contains the following column and information:

- "gene_symbol" contains the symbol of the described gene, using the HUGO Gene Nomenclature
- ENSEMBL_ID contains the stable identifier from the ENSEMBL database
- cell_class is either "malignant" for marker genes specific to malignant population, or "non-malignant" for markers genes specific to non-malignant tissue or "both" for marker genes that can be found in malignant as well as non-malignant tissue but are still informative in respect to the cell type.
- cell_type contains the list of the cell types that are attributed to the marker gene
- DOI contains the list of main publication identifiers supporting the choice of the marker gene
- comment can be empty or contains any additional information

The table GeneticAlterations_metadata.csv contains the following column and information:

- alteration contains the number and portion of the affected chromosome
- gain_loss contains the information regarding the gain or loss of the corresponding genetic alteration
- cell_class is "malignant"
- cell_type contains the list of the malignant cell types that are attributed to the marker gene, either blastemal, stromal, epithelial or NA if none of the three histology is more prone to the described genetic alteration
- DOI contains the list of main publication identifiers supporting the choice of the genetic alteration
- comment can be empty or contains any additional information

#### If known, do you anticipate filing additional pull requests to complete this analysis module?
This module will be used for later validation of the annotations and results from inferCNV. 

#### What is the name of your results bucket on S3?

_Results should be uploaded to your bucket so they are available during review._
_See here for instructions on how to upload to your bucket:_
https://openscpca.readthedocs.io/en/latest/software-platforms/aws/working-with-s3-buckets/

#### What types of results does your code produce (e.g., table, figure)?
2 tables

### Provide directions for reviewers

This section had 2 aims:
- learn how to build the github repository, perform issue, pull request
- gather literature information into a metadata file for later use for validation of the annotations
- 
### Author checklists

_Check all those that apply._
_Note that you may find it easier to check off these items after the pull request is actually filed._

#### Analysis module and review

- [ ] This analysis module [uses the analysis template and has the expected directory structure](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/analysis-modules/).
- [x ] The analysis module `README.md` has been updated to reflect code changes in this pull request.
- [ ] The analytical code is documented and contains comments.
- [ ] Any results and/or plots this code produces have been added to your S3 bucket for review.

#### Reproducibility checklist

- [ ] Code in this pull request has been added to the GitHub Action workflow that runs this module.
- [x] The dependencies required to run the code in this pull request have been added to the analysis module `Dockerfile`.
- [ ] If applicable, the dependencies required to run the code in this pull request have been added to the analysis module conda `environment.yml` file.
- [x] If applicable, R package dependencies required to run the code in this pull request have been added to the analysis module `renv.lock` file.

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
This looks great! I'm going to close #672 and start my review here.

File: analyses/cell-type-wilms-tumor-06/run.sh
Comment (jaclyn-taroni) [inline analyses/cell-type-wilms-tumor-06/run.sh:6]: I'm not sure if we expect everyone who might want to run this module to have access to `bash/parse_yaml.sh`?

File: analyses/cell-type-wilms-tumor-06/Dockerfile
Comment (jaclyn-taroni) [inline analyses/cell-type-wilms-tumor-06/Dockerfile:2]: I had a problem building this locally, but it seems that the base image might be the issue. I will need to investigate a little more.

**Comment by maud-p** [comment]
> Hi @maud-p - I have to head to some meetings, so I will return the feedback I have so far here. 😄
> 
> Some high-level thoughts:
> 
> * I would move to use the data download script as part of your development process because I think it will make it easier for other people (e.g., reviewers) to run your code over the project's lifecycle.
> * Were you able to build this Docker image successfully locally? If so, what kind of machine are you on? I ran into a problem locally, and I think your answers might help us narrow down what's going on here.
> 
> Please let us know if you have any questions. Thank you!

Thank you for your feedback! I will try to answer.

- I will try to move to the data download script, thanks for the advice below!
- I build this Docker image on my machine. It is our group's compute server, when I log into it say: 

"Welcome to Ubuntu 22.04.2 LTS (GNU/Linux 5.15.0-101-generic x86_64)"

  System load:  0.79736328125       Processes:                    1245
  Usage of /:   44.1% of 196.30GB   Users logged in:              0
  Memory usage: 25%             
  Swap usage:   51%                

I used podman to build the image
podman build -t cancerbits/dockr:maudp_ScPCAOpen_podman -f Dockerfile.Dockerfile .

I will try to work on the improvment you suggested below for the Dockerfile, it might also solve the issue.

File: analyses/cell-type-wilms-tumor-06/Dockerfile
Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/Dockerfile:6]: > From experience ([AlexsLemonade/OpenPBTA-analysis#1641](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/1641) 😅 ), I know why you have to set an MRAN snapshot from some time ago, but I'm not sure how nicely it will "play" with the Bioconductor base image using `3.19`.
> 
> If you are concerned about being tied to specific versions, I think one way to approach this would be to use the [`renv` package](https://rstudio.github.io/renv/articles/renv.html) in conjunction with Docker.
> 
> Here are some links to the OpenScPCA docs that you might find helpful:
> 
> * https://openscpca.readthedocs.io/en/latest/ensuring-repro/managing-software/using-renv/
> * https://openscpca.readthedocs.io/en/latest/ensuring-repro/docker/docker-images/#r-based-images
> 
> We can file an issue to add `renv` later, though; it doesn't need to happen in this pull request!

Thank you for pointing to these links! I'll try to build the docker as you suggested, sorry I missed this before!
image

File: analyses/cell-type-wilms-tumor-06/Dockerfile
Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/Dockerfile:17]: Thank you @jaclyn-taroni for your comment on the Dockerfile! FYI, I decided to rebuild an image based on the template you provided here : https://openscpca.readthedocs.io/en/latest/ensuring-repro/docker/docker-images/#r-based-images and try to take your comments into account! 

Also, I will follow your instructions to build the image: 
# move to the module directory
cd /path/to/your-module
# build the docker image
docker buildx build . -t openscpca/your-module:latest --platform linux/amd64

I am quite confident that this will solve the problem of building the docker image. :)

File: analyses/cell-type-wilms-tumor-06/run.sh
Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/run.sh:6]: ... Start being a bit too high-level for me to be honnest... 
All our template have been done by our expert Christoph Hafemeister, and here, I have to admit, I run it without understanding every line... 

I'll ask him once he is back from vacation!

Would this replace the run.sh and config.yaml files in the end?
https://openscpca.readthedocs.io/en/latest/ensuring-repro/docker/docker-images/#r-based-images
image

File: analyses/cell-type-wilms-tumor-06/Dockerfile
Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/Dockerfile:2]: I changed for this:

```shell
# Base image on the Bioconductor 3.19 image
FROM bioconductor/r-ver:3.19
``` 

and build locally like this 
```shell
podman buildx build . -t openscpca/cell-type-wilms-tumor-06:latest --platform linux/amd64
``` 
and it works :)

Comment (jaclyn-taroni) [reply analyses/cell-type-wilms-tumor-06/Dockerfile:2]: I am building locally to make some tested recommendations regarding what to do with `run.sh`, but it is taking a while, so I'll post an update as soon as possible!

File: analyses/cell-type-wilms-tumor-06/README.md
Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/README.md:39]: AWS CLI setup and download into data worked well 👍 
```shell
../../download-data.py --projects SCPCP000006
``` 
I'll change the READ.ME file, config.yaml and (future) scripts!

File: analyses/cell-type-wilms-tumor-06/Dockerfile
Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/Dockerfile:2]: Thank you that would be great! 

I am initiating a renv environment from the current R Session to then simplify the Dockerfile using the renv.lock (https://openscpca.readthedocs.io/en/latest/ensuring-repro/docker/docker-images/)

renv_init() is taking a while... I'll continue tomorrow :)

Comment (jaclyn-taroni) [reply analyses/cell-type-wilms-tumor-06/Dockerfile:2]: I had trouble accessing RStudio on the image I built locally. I'm attempting to isolate the problem... but that unfortunately requires rebuilding things 😅 ⏳

Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/Dockerfile:2]: Thank you for being on it :) I have troubles trying to implement the renv() environment and to change for the bioconductor/r-ver:3.19 based image... So far I can only open RStudio using the following Dockerfile (just a bit "cleaned" compared to the PR).

```shell
# Base image on the Bioconductor 3.19 image
FROM bioconductor/tidyverse:3.19

# Set global R options
RUN echo "options(repos = 'https://cloud.r-project.org')" > $(R --no-echo --no-save -e "cat(Sys.getenv('R_HOME'))")/etc/Rprofile.site
ENV RETICULATE_MINICONDA_ENABLED=FALSE

RUN R --no-echo --no-restore --no-save -e "install.packages('remotes')"

RUN R -e "devtools::install_github('enblacar/SCpubr')"
RUN R -e "remotes::install_github('satijalab/seurat', 'seurat5', quiet = TRUE)"  # this also install patchwork (and others)
RUN R -e "remotes::install_github('satijalab/azimuth', quiet = TRUE)"            # this also install SingleCellExperiment, DT (and others)
RUN R  -e "remotes::install_github('cancerbits/DElegate')"
RUN R  -e "install.packages('viridis')"
RUN R  -e "install.packages('ggplotify')"
RUN R -e "BiocManager::install('edgeR')"


# make sure all R related binaries are in PATH in case we want to call them directly
ENV PATH ${R_HOME}/bin:$PATH
```

Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/Dockerfile:2]: @jaclyn-taroni FYI, I removed from the run.sh et config file unnecessary volume and path to data that are specific to our group. I realized it prevent the execution of the docker image if not defined!

Just in case it can help.

Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/Dockerfile:2]: Last comment of the week ;) 

I added the renv.lock file from the RStudio session of docker image I am running on my machine.

I am now trying to use it to build the image as described here: (https://openscpca.readthedocs.io/en/latest/ensuring-repro/docker/docker-images/), but building of the image is full of ERROR because of BiocManager version not matching the bioconductor version.. Will take some time of fine tunning  ⏳

But I think it's worth trying to have the Docker image in this format, then it might be easier to share/reproduce? What do you think @jaclyn-taroni ?

Comment (jaclyn-taroni) [reply analyses/cell-type-wilms-tumor-06/Dockerfile:2]: I agree that our end goal should be to use `renv` as part of the Docker build process, but I don't know if that's necessarily where we need to start. I think it's fine to start installing packages one-by-one in the Dockerfile, and then view `renv` as a feature enhancement!

I am specifically struggling with what base image should be used right now. 

I expect folks working on this project (including all of our staff at the Data Lab) to be on Macs with Apple Silicon. 

My understanding is that you want to be able to develop using the RStudio Server IDE from within the running container. This is often part of my workflow, and I expect many other project participants might have this use case! (That is to say, I think we are bumping into a problem that will come up again and again in the project...)

However, I'm not sure we can use `bioconductor/tidyverse:3.19` as the base image and have it play nicely with multiple architectures (e.g., if someone using an ARM machine wants to build and use it locally). That's because I don't think `rocker/tidyverse`, which I understand to be the base of `bioconductor/tidyverse`, supports ARM: https://github.com/rocker-org/rocker-versioned2/issues/830

So, we might instead want to use the `bioconductor_docker` as the base:

```
# pull base image
FROM bioconductor/bioconductor_docker:3.19
```

And until we implement `renv`, I expect we can install the Tidyverse packages the same way suggested in that issue ☝🏻 

```
# pull base image
FROM bioconductor/bioconductor_docker:3.19

# Adds tidyverse packages & devtools
RUN /rocker_scripts/install_tidyverse.sh
```

Then I _believe_ we'd have an image we can build and push to Elastic Container Registry using GitHub Actions that can also be built and used locally on ARM machines. This compatibility seems like a good goal to me.

I would have liked to test this all conclusively, but installing Azimuth is taking a very long time for me 😅 Appreciate your patience, @maud-p!

File: analyses/cell-type-wilms-tumor-06/run.sh
Comment (jaclyn-taroni) [inline analyses/cell-type-wilms-tumor-06/run.sh:1]: Can we rename this to `run-podman-internal.sh`, please? `run.sh` might imply that this is how you run all steps in the module, so I think this name is more descriptive.

Did you need to add this file with force, i.e., `git add --force analyses/cell-type-wilms-tumor-06/run.sh`? I would expect this file to be ignored, which is why I ask. 

Generally, you want to avoid using `--force`. There are many things in [the `.gitignore` in the root of the repository](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/0d077d444a2f339ff84a8b5ca0a2eabb1e247350/.gitignore) to prevent us from committing things we want to put in S3 instead, for example, and `--force` allows you to avoid the guardrails.

File: analyses/cell-type-wilms-tumor-06/run.sh
Comment (jaclyn-taroni) [inline analyses/cell-type-wilms-tumor-06/run.sh:1]: I have not tested this (and probably could not satisfactorily without access to your system), so please test it on your end. I think it would be better to pass the password as an argument:

```suggestion
PASSWORD="$1"

# ids defined in image for the rstudio user, if not define as such, it is not possible to login to RStudio
```

I will add what I think needs to happen in line 18 for this to work.

File: analyses/cell-type-wilms-tumor-06/run.sh
Comment (jaclyn-taroni) [inline analyses/cell-type-wilms-tumor-06/run.sh:18]:
```suggestion
  -e PASSWORD=$PASSWORD \
```

File: analyses/cell-type-wilms-tumor-06/README.md
Comment (jaclyn-taroni) [inline analyses/cell-type-wilms-tumor-06/README.md:168]: Can you add an H4 section here on Halbritter lab internal development please?

I'd expect that would include how to run the script (this is taking into account some review feedback):

```
./run-podman-internal.py {PASSWORD}
```

This hopefully helps you with your own development if, for example, you go on vacation for two weeks and come back to this! 😄  But it also helps others understand that this bash script isn't for their use.

File: analyses/cell-type-wilms-tumor-06/README.md
Comment (jaclyn-taroni) [inline analyses/cell-type-wilms-tumor-06/README.md:28]: I don't know if we need this sentence now that the other instructions use the data download script.

```suggestion
```

**Review by jaclyn-taroni — COMMENTED** [review]
@maud-p, this is very close! 🎉

I left a few comments on the README about the marker sets:

- For some of the genes included, I might expect it to be helpful to examine altered expression patterns/programs rather than individual gene expression. You do not need to take any action besides replying with what you think about those ideas—these are just items for scientific discussion and for us to consider moving forward!
- I had a question about why _THY1_ was included vs. other markers based on one publication.
- I had a problem with one of the DOIs that’d be good to check out before merging.

Regarding how you need to run the Docker image:

* I’d rename the `run.sh` to be more general and to accept the password as an argument.
* I’d write a section in the README that explains how you use the script to run the container on your own system.

We now know that the Docker image can be built because of the workflow we added! ✅ I link to the successful run in one of my comments so you can check it out.

I also wanted to mention that your decision to close #680, as mentioned in https://github.com/maud-p/OpenScPCA-analysis/pull/3#issuecomment-2267780226,  and re-open a new pull request makes sense to me!

File: analyses/cell-type-wilms-tumor-06/README.md
Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/README.md:85]: I added the (?) for the reason you mentionned, as we are looking for loss of function, I am not sure that we can really use it for annotation. 

However, about 20% of Wilms tumor would have imperment of _WT1_, and I would expect the _WT1_ mutated Wilms tumor to have a specific transcriptional program. At the final step of integration of the 40 samples together, I would expect a cluster negative for _WT1_. Also, the normal kidney should be _WT1_ positive. So in this last step I think looking at _WT1_ would make sense.

File: analyses/cell-type-wilms-tumor-06/README.md
Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/README.md:87]: This is a great idea, I will implement this in the next PR:
- Differential expression analysis using DElegate (pseudobulk based) FindAllMarkers2 to find markers for each clusters
- Enrichment analysis of the markers genes, I was thinking using the gene sets hallmarks and MSigdB C8, I found them the most informative. The Hallmarks of Cancer might help us defining specific cancer clusters (proliferation +++, DNA damage/repair +++ in anaplastic Wilms tumor). And the MSigdB C8 can be an additionnal level of information regarding cell types. Do you think to some other gene sets?

File: analyses/cell-type-wilms-tumor-06/README.md
Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/README.md:91]: In the MSigdB C3/MIR gene sets, we have gene sets for DICER and DROSHA, I could also give a try to run enrichment for this dataset.

Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/README.md:91]: https://www.gsea-msigdb.org/gsea/msigdb/human/genesets.jsp?collection=MIR

File: analyses/cell-type-wilms-tumor-06/README.md
Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/README.md:87]: https://www.gsea-msigdb.org/gsea/msigdb/human/genesets.jsp?collection=H
https://www.gsea-msigdb.org/gsea/msigdb/human/genesets.jsp?collection=C8 : 50 gene sets kidney related

File: analyses/cell-type-wilms-tumor-06/README.md
Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/README.md:106]: unfortunately I don't know about one mesenchymal gene specific for mesenchymal Wilms tumor cells... For some colleagues who wanted to characterize CAF, the best approach I found was:
1) identify stromal cells and
2) based on the inferedCNV, distinguish normal from cancer stromal cells at the cluster level. 

This is not perfect I think, but at least we should have clusters enriched in the target population. 

Stromal cells are really easily identified based on either few markers, or label transfer from the fetal kidney reference. They often form one single cluster. This is the reason why I didn't spend too much time adding marker genes for them, but I can add few more mesenchymal markers and references for correctness :)

Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/README.md:106]: * and _THY1_ is generally nicely expressed is I remember well!

Comment (jaclyn-taroni) [reply analyses/cell-type-wilms-tumor-06/README.md:106]: I'd defer to you since you've spent more time thinking about this problem 😄 

You don't need to add them – we'll have a record of this conversation!

File: analyses/cell-type-wilms-tumor-06/README.md
Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/README.md:120]: oh sorry, complete doi is https://doi.org/10.1128/mcb.9.4.1799-1803.1989

Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/README.md]: should be corrected in the READ.ME and csv file :)

File: analyses/cell-type-wilms-tumor-06/Dockerfile
Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/Dockerfile:1]: this is a great news, thank you!!!
Should we wait a more complete Docker image before pushing it?
I would like to adapt the notebook with you comment (enrichment analysis of some gene sets) before filling the next pull request, with a docker image (renv.lock) that should be more complete.

Comment (jaclyn-taroni) [reply analyses/cell-type-wilms-tumor-06/Dockerfile:1]: There's no need to wait – incremental changes are okay and expected! As you add R packages you need, I expect the `renv.lock` file to change, which means the Docker image will get rebuilt with the new R packages installed and pushed to ECR. That is to say that the workflow is set up to accommodate incremental changes!

File: analyses/cell-type-wilms-tumor-06/README.md
Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/README.md:168]: I'll have to ask for some help in the Halbritter lab, add it on my to do list ;)

File: analyses/cell-type-wilms-tumor-06/Dockerfile
Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/Dockerfile:1]: great !

**Review by jaclyn-taroni — APPROVED** [review]
@maud-p - I think this is good to go in as is 👍🏻 

Congratulations on your first contribution 🥳 

If you get more information about running this internally or need to make refinements, it's fine to add that in a later pull request!

I'm going to merge the `AlexsLemonade:main` branch into this one to make sure it is up-to-date, which is a prerequisite for merging this into `AlexsLemonade:main`.

Once this goes in, I recommend creating a new branch to add your clustering analyses from `AlexsLemonade:main`. 

Here are some instructions using GitKraken: https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/working-with-git/working-with-branches/#creating-a-feature-branch-in-gitkraken

Side note: If you haven't checked out GitKraken yet, you can use the free version with this project: https://www.gitkraken.com/

I'm not 100% sure how feasible it is to use GitKraken with your lab's setup, but I figured I'd let you know about it.

If you can't use GitKraken, those instructions are less helpful 😅  so if you have any questions, ping me and/or @jashapiro + @sjspielman (Data Lab scientists) in a comment on #679

Thank you, and congratulations again!

**Comment by maud-p** [comment]
Thank you very much!!! I am really happy about it and looking forward the next step of the analysis :)

Thank you for your help setting up all of this!

**Pull request merged**
